### PR TITLE
add training losses to learning rates file

### PR DESCRIPTION
**Pull request by JackTemaki**
Also @Icemole @patrick-wilken, I would prefer to write `train_loss_<name>` and `dev_loss_<name>` instead of `score`, what is your opinion on that?

### CODE REVIEW

**Comment by albertz** [comment]
Note the failing tests. Not sure if you want to fix that in here or separately.

**Comment by albertz** [comment]
Note that `LearningRateControl.get_error_key` (and maybe other places) assume "score", not "loss", and thus they break. But break silently, i.e. the heuristics will not work and will return some other things instead, in certain cases. This might be very unexpected, esp. when people switch from TF to PT. So I would extend at least `get_error_key` to also behave the same for "loss". And check if there are other places where we assume "score".

**Comment by JackTemaki** [comment]
> > Note the failing tests. Not sure if you want to fix that in here or separately.
> 
> I do not understand at all why it fails now and not before. Why should it have been able to write models to the same path but not the learning_rates file now?

My first solution would be now to just create the dir within the config, but this does not feel right...

**Comment by albertz** [comment]
Can you maybe rebase?

**Comment by JackTemaki** [comment]
> > but rather enforce that this is set explicitly for the torch backend if necessary.
> 
> Note that this currently is not really possible. Even when setting `learning_rate_control_error_measure` in the config, this will still be used as part of the heuristics, and might not need to match exactly, and if not found, it will fall back to the other heuristics. This is actually #1146.

Yes but it can be made possible. But lets continue with this on #1146, I forgot this existed.

**Review by albertz — APPROVED** [review]
For me, this is fine.

I would like if we do this for TF as well, but this can be a separate commit/PR.

I think @patrick-wilken or @Icemole should maybe also comment.

**Comment by Icemole** [comment]
I haven't been following the discussion that much as I haven't had the time to look into anything, but I also like these changes and I actually prefer the term `loss` rather than the more ambiguous term `score`.

I assume the assumptions about `score...` that you discussed have already been taken care of?

**Review by Icemole — APPROVED** [review]

**Comment by albertz** [comment]
> I assume the assumptions about `score...` that you discussed have already been taken care of?

Not sure. I did not check this. But we should check for other usages of "score" somewhere. I just pointed to `get_error_key`.

**Comment by patrick-wilken** [comment]
"loss" instead of "score" because that's what we minimize, right? Agreed.

I don't find usages of `train_score` / `dev_score` strings outside of the LR control code.

**Review by patrick-wilken — APPROVED** [review]
